### ISSUE TITLE
opengrok history and annotation links are not working

### ISSUE THREAD

**Issue by ssaadon** [OpenGrok/OpenGrok#1021]
Hi,

We are using OpenGrok 0.9 . Last week we have upgraded our subversion to version 1.92 .
Ever since, history and annotation links are not working.
We are using Java 1.6 .
Log files seems ok:
INFO: Creating historycache for D:\Software\opengrok\working-copies\3.5 took (54966ms)
Nov 1, 2015 12:56:11 PM org.opensolaris.opengrok.history.HistoryGuru createCache

If there is anything else which is missing please let me know.

Thanks!

**Comment by vladak** [comment]
0.9 is really old. Could you upgrade to 0.12 ?

**Comment by moisei** [comment]
We can not yet, working it but it will take time.
Is it possible to trace down this issue - why the links became disabled?
The history is indexed properly - no error messages.

**Comment by ssaadon** [comment]
Guys, please advise.

**Comment by tarzanek** [comment]
Downgrade to old svn?
Or try upgrading OpenGrok ...

**Comment by ssaadon** [comment]
Hello again,
We tried to use Opengrok with docker.
https://hub.docker.com/r/itszero/opengrok/
They  are using the latest openGrok version, but still we are unable to use 'history' tag.

Please let me know which info is required to trace this bug.

Thanks,
Shuki.

**Comment by tarzanek** [comment]
first thing is to look at logs inside /var/opengrok/log
and see what is going on
OpenGrok detects repositories first
also note that user that checks out svn has to have permissions to check it out and ideally the same user has to run the webapp container
If there is any error, the loggers would kinda show it, so read them carefully ...

**Comment by tarzanek** [comment]
second part are the webapp logs, which will also point out any prob with running commands against the repo
third thing to note is the whether the data directory will have the history cache created

**Comment by ssaadon** [comment]
Hello,
Thanks for your reply.

1. log files seems fine:
2015-11-30 09:27:09.150+0000 INFO t1 Statistics.report: Done historycache for all repositories (took 34 ms)

2. user that checks out svn  have permissions to check it, but it's not  same user who run the webapp container - if this is critical I can change that.

3. I found following warning inside tomcat logs:
Nov 30, 2015 9:37:24 AM org.opensolaris.opengrok.history.RepositoryFactory getRepository
WARNING: SubversionRepository not working (missing binaries?): /src


4. history cache is created:
docker@dev:/mnt/sda1/var/lib/docker/aufs/mnt/c4f40c2e8204d1d79955d929a2f63bb488882149b963d3fd61368cd172e60950/grok/data$ ll historycache/
total 56
-rw-r--r--    1 root     root             6 Nov 29 14:22 OpenGroklatestRev
-rw-r--r--    1 root     root           326 Nov 29 14:22 builds.js.gz
-rw-r--r--    1 root     root           331 Nov 29 14:22 datastores.js.gz
-rw-r--r--    1 root     root           338 Nov 29 14:22 hosts.js.gz
-rw-r--r--    1 root     root           558 Nov 29 14:22 index.js.gz
-rw-r--r--    1 root     root           526 Nov 29 14:23 install-dalet.js.gz
-rw-r--r--    1 root     root           492 Nov 29 14:22 inventory.js.gz
-rw-r--r--    1 root     root           323 Nov 29 14:22 logs.js.gz
-rw-r--r--    1 root     root           471 Nov 29 14:22 monitor.js.gz
-rw-r--r--    1 root     root           332 Nov 29 14:22 settings.js.gz
-rw-r--r--    1 root     root           798 Nov 29 14:22 sites.js.gz
-rw-r--r--    1 root     root           344 Nov 29 14:22 templates.js.gz
-rw-r--r--    1 root     root           579 Nov 29 14:22 users.js.gz
-rw-r--r--    1 root     root           749 Nov 29 14:22 virtual-machines.js.gz

Thanks!

**Comment by tarzanek** [comment]
OK
so 2) and 3) are the problem
first try to fix 3) , make sure tomcat has in its path the path to svn binary
and then try to fix 2) so tomcat will be able to execute svn on top of appropriate repo

xing the fingers
L

**Comment by ssaadon** [comment]
It seems like svn is in tomcat's PATH:
root@dev:/mnt/sda1/var/lib/docker/aufs/mnt/c4f40c2e8204d1d79955d929a2f63bb488882149b963d3fd61368cd172e60950/usr/share/tomcat7/bin# docker exec c4f40c2e8204 sudo -u  tomcat7 echo $PATH
/usr/local/sbin:/usr/local/bin:/apps/bin:/usr/sbin:/usr/bin:/sbin:/bin

here is the PATH for svn:
root@dev:/mnt/sda1/var/lib/docker/aufs/mnt/c4f40c2e8204d1d79955d929a2f63bb488882149b963d3fd61368cd172e60950/usr/share/tomcat7/bin# docker exec c4f40c2e8204 sudo which svn
/usr/bin/svn

If I need to configure this somewhere else, please let me know.

I also found this topic, in which they are explicitly stating that subversion cannot run with tomcat 

http://www.wandisco.com/svnforum/forum/opensource-subversion-forums/general-setup-and-troubleshooting/windows-servers/4332-subversion-using-tomcat


Regarding 2) should I checkout with tomcat7 user or with docker user?

**Comment by tarzanek** [comment]
hmm, don't know, never played with docker, but I suppose tomcat7 user runs tomcat, so tomcat7 user should be able to work with svn
I think we run svn from tomcat, do we @vladak ? And it works afaik

**Comment by vladak** [comment]
Yep, svn works fine in our 0.12 setup.

**Comment by moisei** [comment]
Hello,
we've finally resolved this issue, here it is:
Opengrok calls to svn history without user and password. In this case svn
client is using last cached credentials to access the repository.
Obviously this credentials are not set in docker environment and then svn
attempts to connect to the respository anonymously.
Our repo doesn't allow anonymous access, so svn history was not available
to grok.

Once we managed to update subversion crdentials cache on the docker
instance, the issue has been resolved.

Thank you for your help & Thank to Chien-An Cho (in BCC of this email) for
great docker opengrok container.

Best Regards,
Shuki and Moisei.

On Wed, Dec 2, 2015 at 12:05 PM, Vladimir Kotal <notifications@github.com>
wrote:

> Yep, svn works fine in our 0.12 setup.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenGrok/OpenGrok/issues/1021#issuecomment-161245713>.
>



-- 
Best Regards,
Moisei
